src/routes/rooms.ts

```diff
@@ -49,10 +49,18 @@ router.post(
   asyncHandler(async (req, res) => {
     const body: CreateRoom = req.body;
 
-    if (!body || Array.from(body.options).length < 1) {
-      res.status(HttpStatus.UNPROCESSABLE_ENTITY);
-      res.send();
-      return;
+    if (!body) {
+      throw new UnprocessableEntityError(
+        'ROOM_CREATE_EMPTY_DATA',
+        'Empty body sent',
+      );
+    }
+
+    if (!body.options || Array.from(body.options).length < 1) {
+      throw new UnprocessableEntityError(
+        'ROOM_CREATE_EMPTY_OPTIONS',
+        'Options cannot be empty',
+      );
```

Comment:
I would've split this change into a different PR.
Seems like your PR title and description has nothing to do with this.

Reply:
Yes, by adding tests I found this bug and took the liberty of including it here just because I wanted all the introduced tests to pass, but true, I should have created a separate PR for it (or at least I should've written a better PR title + description 😅)